### PR TITLE
Try pulling docker image on scargo update

### PR DESCRIPTION
**Pull request by jze-spyro**
#152

### CODE REVIEW

File: scargo/commands/update.py [hunk 1]
```diff
@@ -95,6 +95,26 @@ def scargo_update(config_file_path: Path) -> None:
 
     if project_config.build_env == SCARGO_DOCKER_ENV:
         if not Path(project_path, ".dockerenv").exists():
-            scargo_docker_build([])
+            pulled_image = False
+            try:
+                logger.info("Pulling the image from docker registry...")
+                result = subprocess.run(
+                    ["docker-compose", "pull"],
```
Comment (aaksenczuk) [inline scargo/commands/update.py:102]: docker instead of docker-compose

Comment (jze-spyro) [reply scargo/commands/update.py:102]: why?

File: tests/ut/ut_scargo_update.py [hunk 2]
```diff
@@ -36,7 +36,21 @@ def test_update_project_content_with_docker(tmp_path: Path, fp: FakeProcess) ->
     os.chdir(tmp_path)
     project_name = "test_project_with_docker"
     scargo_new(project_name, None, None, TARGET_X86, True, False)
+    fp.register("docker-compose pull")
```
Comment (aaksenczuk) [inline tests/ut/ut_scargo_update.py:39]: docker instead of docker-compose

File: tests/ut/ut_scargo_update.py [hunk 3]
```diff
@@ -36,7 +36,21 @@ def test_update_project_content_with_docker(tmp_path: Path, fp: FakeProcess) ->
...
+        assert path.name in EXPECTED_FILES_AND_DIRS
+
+
+def test_update_project_content_with_docker__build(
+    tmp_path: Path, fp: FakeProcess
+) -> None:
+    os.chdir(tmp_path)
+    project_name = "test_project_with_docker"
+    scargo_new(project_name, None, None, TARGET_X86, True, False)
+    fp.register("docker-compose pull", returncode=1)
```
Comment (aaksenczuk) [inline tests/ut/ut_scargo_update.py:51]: docker instead of docker-compose

File: tests/ut/ut_scargo_update.py (hunk 2 above)
Comment (jze-spyro) [reply tests/ut/ut_scargo_update.py:39]: why?

File: tests/ut/ut_scargo_update.py (hunk 3 above)
Comment (jze-spyro) [reply tests/ut/ut_scargo_update.py:51]: why?

File: scargo/commands/update.py [hunk 4]
```diff
@@ -95,6 +95,26 @@ def scargo_update(config_file_path: Path) -> None:
 
     if project_config.build_env == SCARGO_DOCKER_ENV:
         if not Path(project_path, ".dockerenv").exists():
-            scargo_docker_build([])
+            pulled_image = False
```
Comment (aaksenczuk) [inline scargo/commands/update.py:98]: this is not the best place in my opinion - build should always build the image on demand, 
we want to pull docker image only for scargo update and if image not exist than build

Comment (jze-spyro) [reply scargo/commands/update.py:98]: And what makes you think that build will not always build the image on demand?

File: scargo/commands/update.py (hunk 1 above)
Comment (aaksenczuk) [reply scargo/commands/update.py:102]: ok

**Review by aaksenczuk — APPROVED** [review]

**Pull request merged**